provider/dyn.go

```diff
@@ -34,9 +35,6 @@ import (
 const (
 	// 10 minutes default timeout if not configured using flags
 	dynDefaultTTL = 600
-	// can store 20000 entries globally, that's about 4MB of memory
```

Comment:
I would keep the cache - if a zone hasn't been modified we'd save Dyn a lot of work returning probably a big list of record. For external-dns this would mean we can use a lower refresh interval.

Reply:
This seemed redundant to me. All records for a zone/serial are stored using the ZoneSnapshot struct:
`d.ZoneSnapshot.StoreRecordsForSerial(zone, serial, relevantRecords)` 
We only fetch records if the serial number changes, if it doesn't increase, we use whatever is stored in the **ZoneSnapshot** struct (which is already acting as a cache).
Am I missing something?